index.d.ts

```diff
@@ -212,122 +247,137 @@ export interface ContentType {
+    widgetNamespace: "sidebar-builtin" | "extension",
+    widgetId: string,
+    settings: ISidebarWidgetSettings
+  ): void;
 
   /**
    * Removes a widget from the sidebar of the content type.
    *
    * @param widgetNamespace The namespace of the widget. Use 'sidebar-builtin' for standard widgets or 'extension' for UI extensions.
    * @param widgetId The ID of the widget to remove.
    */
-  removeSidebarWidget (widgetNamespace: 'sidebar-builtin' | 'extension',
-                       widgetId: string): void
+  removeSidebarWidget(
+    widgetNamespace: "sidebar-builtin" | "extension",
+    widgetId: string
+  ): void;
 
   /**
    * Resets the sidebar of the content type to default
    */
-  resetSidebarToDefault (): void
+  resetSidebarToDefault(): void;
 }
 
 export interface IContentTypeOptions {
   /** Name of the content type. */
-  name: string,
+  name: string;
   /** Description of the content type. */
-  description?: string,
+  description?: string;
   /** ID of the field to use as the display field for the content type. */
-  displayField?: string
+  displayField?: string;
 }
 
-type ContentFields = { [field: string]: { [locale: string]: any } }
+type ContentFields = { [field: string]: { [locale: string]: any } };
 
 export interface ITransformEntriesConfig {
   /** (required) – Content type ID */
-  contentType: string,
+  contentType: string;
   /** (required) – Array of the source field IDs */
-  from: string[],
+  from: string[];
   /** (required) – Array of the target field IDs */
-  to: string[],
+  to: string[];
   /**
    * (required) – Transformation function to be applied.
    *
    * fields is an object containing each of the from fields. Each field will contain their current localized values (i.e. from == {myField: {'en-US': 'my field value'}})
    * locale one of the locales in the space being transformed
    *
    * The return value must be an object with the same keys as specified in to. Their values will be written to the respective entry fields for the current locale (i.e. {nameField: 'myNewValue'}). If it returns undefined, this the values for this locale on the entry will be left untouched.
-  */
-  transformEntryForLocale: (fromFields: ContentFields, currentLocale: string) => any,
+   */
+  transformEntryForLocale: (
+    fromFields: ContentFields,
+    currentLocale: string
+  ) => any;
   /** (optional) – If true, the transformed entries will be published. If false, they will remain in draft state. When the value is set to "preserve" items will be published only if the original entry was published as well (default true) */
-  shouldPublish?: boolean|"preserve"
+  shouldPublish?: boolean | "preserve";
 }
 
 export interface ITransformEntriesToTypeConfig {
   /** (required) – Content type ID */
-  contentType: string,
+  sourceContentType: string;
+  /** (required) – Content type ID */
+  targetContentType: string;
   /** (required) – Array of the source field IDs */
-  from: string[],
+  from: string[];
```

Comment:
Since `from` is marked as optional in the [docs](https://github.com/contentful/contentful-migration/blob/master/README.md#transformentriestotypeconfig):

```suggestion
  from?: string[];
```